pytests/tests/test_signature.py

```diff
@@ -0,0 +1,90 @@
+#
+# Copyright (C) 2019 - 2020 VMware, Inc. All Rights Reserved.
+#
+# Licensed under the GNU General Public License v2 (the "License");
+# you may not use this file except in compliance with the License. The terms
+# of the License are located in the COPYING file of this distribution.
+#
+#   Author: Siddharth Chandrasekaran <csiddharth@vmware.com>
+
+import os
+import re
+import tempfile
+import pytest
+
+DIST = os.environ.get('DIST')
+if DIST == 'photon':
+    DEFAULT_KEY='file:///etc/pki/rpm-gpg/VMWARE-RPM-GPG-KEY'
+elif DIST == 'fedora':
+    DEFAULT_KEY='file:///etc/pki/rpm-gpg/RPM-GPG-KEY-fedora-34-primary'
+
+@pytest.fixture(scope='function', autouse=True)
+def setup_test_function(utils):
+    utils.run(['rpm', '-e', '--allmatches', 'gpg-pubkey'])
+    pkgname = utils.config["sglversion_pkgname"]
+    utils.run(['tdnf', 'erase', '-y', pkgname])
+    yield
+    teardown_test(utils)
+
+def teardown_test(utils):
+    set_gpgcheck(utils, False)
+    utils.run(['rpm', '-e', '--allmatches', 'gpg-pubkey'])
+    pkgname = utils.config["sglversion_pkgname"]
+    utils.run(['tdnf', 'erase', '-y', pkgname])
+
+def set_gpgcheck(utils, enabled):
+    tdnf_repo = os.path.join(utils.tdnf_config.get('main', 'repodir'), 'photon-test.repo')
+    if enabled:
+        utils.run([ 'sed', '-i', '/gpgcheck/s/.*/gpgcheck=1/g', tdnf_repo ])
+    else:
+        utils.run([ 'sed', '-i', '/gpgcheck/s/.*/gpgcheck=0/g', tdnf_repo ])
+
+def set_repo_key(utils, url):
+    tdnf_repo = os.path.join(utils.tdnf_config.get('main', 'repodir'), 'photon-test.repo')
+    utils.run(['sed', '-i', '/gpgkey/s/.*/#gpgkey=/g', tdnf_repo])
+    if url != None:
+        utils.run(['sed', '-i', '/gpgkey/s@.*@gpgkey={}@g'.format(url), tdnf_repo])
+
+# 'wrong' key in repo config, expect fail
+def test_install_nokey(utils):
+    set_gpgcheck(utils, True)
+    set_repo_key(utils, DEFAULT_KEY)
+    pkgname = utils.config["sglversion_pkgname"]
+    ret = utils.run([ 'tdnf', 'install', '-y', pkgname])
+    assert(ret['retval']  == (1514 & 0xff))
```

Comment:
This is no longer needed. You can check for full return codes.

Reply:
Nice! Fixed.